components/idenpubonchain/idenpubonchain.go

```diff
@@ -162,16 +163,22 @@ func splitSignature(signature *babyjub.SignatureComp) (sigR8 [32]byte, sigS [32]
 }
 
 // SetState updates the Identity State of the given ID in the IdenStates Smart Contract.
-func (ip *IdenPubOnChain) SetState(id *core.ID, newState *merkletree.Hash, kOpProof []byte,
+func (ip *IdenPubOnChain) SetState(id *core.ID, newState *merkletree.Hash, kOp babyjub.PublicKey,
```

Comment:
`components/idenpubonchain.IdenPubOnChain` and `components/idenpubonchain.local.IdenPubOnChain` should always fulfill the interface `components/idenpubonchain.IdenPubOnChainer`.

But the `SetState` and `InitState` of `components/idenpubonchain.IdenPubOnChain` doesn't follow the interface.

Please, update the interface and `components/idenpubonchain.local.IdenPubOnChain`

Reply:
I have made a PR that adds tests that will catch these things: https://github.com/iden3/go-iden3-core/pull/341

Reply:
Right, done now to fulfill the current interface.
This will be affected and updated in the next phase, as the identity state update methods will be done through a zkproof as commented in this issue https://github.com/iden3/contracts/issues/26